src/Server/TCPHandler.cpp

```diff
@@ -672,6 +685,22 @@ void TCPHandler::receiveUnexpectedTablesStatusRequest()
     throw NetException("Unexpected packet TablesStatusRequest received from client", ErrorCodes::UNEXPECTED_PACKET_FROM_CLIENT);
 }
 
+void TCPHandler::sendFingerprints()
```

Comment:
Again, confusing names. `sendFingerprints` sends fingerprints (all available?) and `receiveFingerprints` receives fingerprints we have to exclude?

Reply:
>sendFingerprints sends fingerprints (all available?)

it's still unclear what the path forward we're going to pick: 
* set fingerprints to only movings parts
* set fingerprints to all parts

current approach is rely on first decision: there should be little amount of parts with fingerprints that were processed during the read.

>and receiveFingerprints receives fingerprints we have to exclude?

agree a bit confusing, can be at least `receiveExcludedFingerprints`